### ISSUE TITLE
[Bug]: changed images in the web interface do not show (plus fix suggestion!)

### ISSUE THREAD

**Issue by mrAceT** [nextcloud/server#40230]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Nextcloud nicely generates images to be shown in the web interface, also there is an image editor available. When one changes something in an image, the preview doesn't seem to update..

### Steps to reproduce

1. upload an image
2. change it via the web interface (add some text for instance)
3. preview image does not seem to change

### Expected behavior

One would expect to see the changed image..

**FIX SUGGESTION**
I have this problem in my coding sometimes too. The problem is the caching of the browser. There is a little trick to fix this:
The image URL is something like:

[URL]/core/preview?fileId=964&x=1920&y=1080&a=true

When one would add something unique, like the file time ('storage_mtime' in 'oc_filecache' would seem to be perfectly suitable for this) then the URL becomes a new one and one can force an update. Something like:

[URL]/core/preview?fileId=964&x=1920&y=1080&a=true&**mtime=1674741268**

Then that URL gets cached (do not add a random number, that would strain the caching unnecessarily)

### Installation method

None

### Nextcloud Server version

25

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by szaimen** [comment]
Cc @nextcloud/server-frontend

**Comment by artonge** [comment]
Indeed, here is an example how the Photos app tackled this: https://github.com/nextcloud/photos/blob/master/src/components/File.vue#L213

**Comment by skjnldsv** [comment]
The Viewer needs to trigger the right event and we need to update the etag accordingly then.

**Comment by artonge** [comment]
It already does with updated data: https://github.com/nextcloud/viewer/blob/master/src/components/ImageEditor.vue#L184-L197
